presto-main/src/test/java/com/facebook/presto/operator/TestPartitionedOutputOperator.java

```diff
@@ -0,0 +1,274 @@
+import java.util.OptionalInt;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.block.BlockAssertions.createLongsBlock;
+import static com.facebook.presto.execution.buffer.BufferState.OPEN;
+import static com.facebook.presto.execution.buffer.BufferState.TERMINAL_BUFFER_STATES;
+import static com.facebook.presto.execution.buffer.OutputBuffers.BufferType.PARTITIONED;
+import static com.facebook.presto.execution.buffer.OutputBuffers.createInitialEmptyOutputBuffers;
+import static com.facebook.presto.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestPartitionedOutputOperator
+{
+    private static final int PARTITION_COUNT = 512;
+    private static final DataSize MAX_MEMORY = new DataSize(1, GIGABYTE);
+    private static final List<Type> TYPES = ImmutableList.of(BIGINT);
+    private static final Random RANDOM = new Random(0);
+    private static final ExecutorService EXECUTOR = newCachedThreadPool(daemonThreadsNamed("test-EXECUTOR-%s"));
+    private static final ScheduledExecutorService SCHEDULER = newScheduledThreadPool(1, daemonThreadsNamed("test-%s"));
+    private static final DataSize PARTITION_MAX_MEMORY = new DataSize(5, MEGABYTE);
+    private static final int POSITION_COUNT = 1000;
+    private static final int NUM_OF_ITERATIONS = 100;
+
+    @Test
+    public void testAddInputForSimplePage()
+    {
+        PartitionedOutputOperator partitionedOutputOperator = createPartitionedOutputOperator(false);
+        for (int i = 0; i < NUM_OF_ITERATIONS; i++) {
+            partitionedOutputOperator.addInput(createSimplePage());
+        }
+        partitionedOutputOperator.finish();
+        OperatorContext operatorContext = partitionedOutputOperator.getOperatorContext();
+        long outputSize = operatorContext.getOutputDataSize().getTotalCount();
+        long outputPositionCount = operatorContext.getOutputPositions().getTotalCount();
+        assertEquals(outputSize, 900_000);
+        assertEquals(outputPositionCount, 100_000);
+        assertEquals(createSimplePage().getSizeInBytes() * NUM_OF_ITERATIONS, outputSize);
```

Comment:
This rely on the fact that every page created has the same size. But by looking at the function name createSimplePage() we can't tell if the size is fixed. I think it's better to accumulate the page sizes inside of the for loop using addition.